### DISCUSSION TITLE
Query list of posts grouped by category

### DISCUSSION THREAD

**Discussion by joaolandino** [prisma/prisma#18658] [Q&A] [answered]
Hi guys!

I need to make a query that creates a list of posts grouped by category with pagination, but I don't know how to do that. Can anyone help me with this?
I created an illustration to be more simpler to explain.
<img width="538" alt="image" src="https://user-images.githubusercontent.com/26358607/230490548-05fb0ad4-0ca6-4cc3-949a-022b0c37c045.png">

Here is my schema:

```
model Collection {
  id         String   @id @default(cuid())
  title      String
  posts      Post[]

  @@map("collections")
}

model Post {
  id            String      @id @default(cuid())
  legend        String
  collection_id String?
  collection    Collection? @relation(fields: [collection_id], references: [id])

  @@map("posts")
}
```

Any suggestions?

Thank you very much!!!

**Comment by nurul3101** [comment, marked as answer]
Hello @joaolandino 👋 

I am assuming that the Collection model represents categories in this example. And pagination is at the individual categories level.
Considering that you want to fetch 4 posts in each page of a collection, your query could look something like this

```typescript
const postsForCategoriesA = await prisma.post.findMany({
  take: 4,
  where: {
    collection_id: '__collection_id_of_A__'
  },
  orderBy: {
    id: 'asc',
  },
})
```
 
Once you get the first 4 records, you would need to store the id of the last fetched post that will act as a cursor.
```ts
const lastPostInResults = postsForCategoriesA[3]
const myCursor = lastPostInResults.id
```

Now, to get the next set of posts for category A, it could look something like:
```typescript
const nextPostsForCategoriesA = await prisma.post.findMany({
  take: 4,
  skip: 1,
   cursor: {
    id: myCursor,
  },
  where: {
    collection_id: '__collection_id_of_A__'
  },
  orderBy: {
    id: 'asc',
  },
})
```

Here's a pagination reference guide: [Docs](https://www.prisma.io/docs/concepts/components/prisma-client/pagination)
This would be helpful to you.

You can repeat this process for fetching posts of other categories as well.

**Reply by joaolandino** [reply]
Thank you very much @nurul3101 !

**Reply by nurul3101** [reply]
No worries! 🙂
